src/irc/core/irc-servers.c

```diff
@@ -78,6 +78,9 @@ static int ischannel_func(SERVER_REC *server, const char *data)
 	IRC_SERVER_REC *irc_server = (IRC_SERVER_REC *) server;
 	char *chantypes, *statusmsg;
 
+	g_return_val_if_fail(data != NULL, FALSE);
+	g_return_val_if_fail(*data != '\0', FALSE);
```

Comment:
I don't think that macro is being used correctly here, see https://developer.gnome.org/glib/stable/glib-Warnings-and-Assertions.html#g-return-val-if-fail

fail means precondition check so this is a programmer error, but the irssi code is legally calling the function like this